invoices/invoiceregistry_test.go

```diff
@@ -240,6 +241,69 @@ func TestSettleInvoice(t *testing.T) {
 	}
 }
 
+// Tests that invoice is rejected after expiration.
+func TestInvoiceExpiry(t *testing.T) {
+	registry, cleanup := newTestContext(t)
+	defer cleanup()
+
+	allSubscriptions := registry.SubscribeNotifications(0, 0)
+	defer allSubscriptions.Cancel()
+
+	const testExpiry = time.Second
+
+	expiringInvoice := &channeldb.Invoice{
+		Terms: channeldb.ContractTerm{
+			PaymentPreimage: preimage,
+			Value:           lnwire.MilliSatoshi(100000),
+		},
+		CreationDate: time.Now(),
+		Expiry:       testExpiry,
+	}
+
+	// Add the invoice.
+	addIdx, err := registry.AddInvoice(expiringInvoice, hash)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	if addIdx != 1 {
+		t.Fatalf("expected addIndex to start with 1, but got %v",
+			addIdx)
+	}
+
+	// We expect a new invoice notification to be sent out.
+	select {
+	case newInvoice := <-allSubscriptions.NewInvoices:
+		if newInvoice.Terms.State != channeldb.ContractOpen {
+			t.Fatalf("expected state ContractOpen, but got %v",
+				newInvoice.Terms.State)
+		}
+	case <-time.After(testTimeout):
+		t.Fatal("no update received")
+	}
+
+	hodlChan := make(chan interface{}, 1)
+
+	// Sleep just enough to expire the invoice.
+	time.Sleep(testExpiry + time.Millisecond)
```

Comment:
Alternatively, perhaps we can just make an invoice that has already expired? So craft the creation date and expiry to already have expired by the time the invoice has been created.